WordPress/gutenberg
# Add @wordpress/navigation package

## What?
This adds a new navigation package, so that we can share code between the block library and the site editor.

## Why?
To be able to achieve things like https://github.com/WordPress/gutenberg/issues/51805, without duplicating loads of code.

## How?
For now this just moves a few useful things to the package, but going forward we should migrate more things there as they become shared.

## Testing Instructions
1. Add a navigation block to a post
2. Check that you can still switch navigations using the navigation selector in the inspector controls.
